### PR TITLE
Write name to LtiDeployment

### PR DESCRIPTION
**Pull request by nicklathe**
This PR updates the logic to write `name` to new LtiDeployments, and update existing LtiDeployments with `name` if empty.

These two cases will handle new deployments, and backfill existing deployments. This work is to further enable better reporting on Schoology installations, as LtiDeployment is used as the unique identifier for Schoology installations.

## Links

[Jira](https://codedotorg.atlassian.net/browse/P20-1223)

## Testing story

```
bundle exec spring testunit ./test/controllers/lti_v1_controller_test.rb
Running via Spring preloader in process 1993
Started with run options --seed 38258

Since there is no EDITOR or BETTER_ERRORS_EDITOR environment variable, using Textmate by default.================================                                             ] 73% Time: 00:00:08,  ETA: 00:00:03
  52/52: [===================================================================================================================================================================] 100% Time: 00:00:11, Time: 00:00:11

Finished in 11.16319s
52 tests, 131 assertions, 0 failures, 0 errors, 0 skips
```
```
bundle exec spring testunit ./test/lib/services/lti_test.rb
Running via Spring preloader in process 3681
Started with run options --seed 28435

  28/28: [===================================================================================================================================================================] 100% Time: 00:00:05, Time: 00:00:05

Finished in 5.18332s
28 tests, 78 assertions, 0 failures, 0 errors, 0 skips
```

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/app/controllers/lti_v1_controller.rb [hunk 1]
```diff
@@ -153,11 +153,20 @@ def authenticate
       nrps_url = decoded_jwt[Policies::Lti::LTI_NRPS_CLAIM]&.[](:context_memberships_url)
       resource_link_id = decoded_jwt[Policies::Lti::LTI_RESOURCE_LINK_CLAIM]&.[](:id)
       deployment_id = decoded_jwt[Policies::Lti::LTI_DEPLOYMENT_ID_CLAIM]
+      deployment_name = decoded_jwt[Policies::Lti::LTI_DEPLOYMENT_PLATFORM_CLAIM]&.[](:name)
       deployment = Queries::Lti.get_deployment(integration[:id], deployment_id)
       lti_account_type = Policies::Lti.get_account_type(decoded_jwt[Policies::Lti::LTI_ROLES_KEY])
 
+      # If deployment name is nil, update it with the name from the JWT. This
+      # could likeyl be removed after a period of time, as we also write the name
```
Comment (nicklathe) [inline dashboard/app/controllers/lti_v1_controller.rb]: Typo. Will fix.

File: dashboard/app/controllers/lti_v1_controller.rb [hunk 2]
```diff
@@ -153,11 +153,21 @@ def authenticate
...
+      deployment_name = decoded_jwt[Policies::Lti::LTI_DEPLOYMENT_PLATFORM_CLAIM]&.[](:name)
       deployment = Queries::Lti.get_deployment(integration[:id], deployment_id)
       lti_account_type = Policies::Lti.get_account_type(decoded_jwt[Policies::Lti::LTI_ROLES_KEY])
 
+      # If deployment name is nil, update it with the name from the JWT. This
+      # could likely be removed after a period of time, as we also write the name
+      # for all new deployments in the next block. This block is necessary to
+      # backfill existing deployments that were created before we started
+      # writing the name.
+      if deployment && deployment.name.nil?
```
Comment (carl-codeorg) [inline dashboard/app/controllers/lti_v1_controller.rb]: Minor nit: could be simplified to `unless deployment&.name`

**Review by carl-codeorg — APPROVED** [review]
One small non-blocker, otherwise LGTM

File: dashboard/app/controllers/lti_v1_controller.rb (hunk 2 above)
Comment (nicklathe) [reply dashboard/app/controllers/lti_v1_controller.rb]: This works if using a safe navigation operator on the update too, i.e. `deployment&.update(...)`.

Copilot recommend this one-liner, which I just pushed to this PR: `deployment&.update(name: deployment_name) if deployment&.name.nil?` If this isn't as readable, I'm happy to change it to your original suggestion.

Comment (carl-codeorg) [reply dashboard/app/controllers/lti_v1_controller.rb]: LGTM, basically the same thing just with `if` instead of `unless`

**Pull request merged**